From 5dd8bd4fa4c6d6220918e18353638b3b3a017026 Mon Sep 17 00:00:00 2001
From: Gowtham Tammana <g-tammana@ti.com>
Date: Fri, 9 Jul 2021 22:53:26 -0500
Subject: [PATCH 1/2] HACK: dt-bindings: misc: Add ti,dma_buf_phys bindig doc

Driver allowing user-space attaching of dmabuf fd's returning CPU
physical addresses. The reasoning for making this act like a regular
device described by DT is so the virtual device that binds the buffer
can be made to act as if it is out on a bus or behind an IOMMU, for
example.

Add this binding here. This is a hack and marked as such.

Signed-off-by: Gowtham Tammana <g-tammana@ti.com>
---
 .../bindings/misc/ti,dma_buf_phys.yaml        | 32 +++++++++++++++++++
 1 file changed, 32 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/misc/ti,dma_buf_phys.yaml

diff --git a/Documentation/devicetree/bindings/misc/ti,dma_buf_phys.yaml b/Documentation/devicetree/bindings/misc/ti,dma_buf_phys.yaml
new file mode 100644
index 000000000000..998bb2943e00
--- /dev/null
+++ b/Documentation/devicetree/bindings/misc/ti,dma_buf_phys.yaml
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+# Copyright 2021 Texas Instruments Incorporated
+%YAML 1.2
+---
+$id: "http://devicetree.org/schemas/misc/ti,dma_buf_phys.yaml#"
+$schema: "http://devicetree.org/meta-schemas/core.yaml#"
+
+title: DMA-BUF contiguous buffer physical address user-space exporter
+
+maintainer: Gowtham Tammana <g-tammana@ti.com>
+
+description: |
+  Driver allowing user-space attaching of DMA-BUFs returning CPU
+  physical addresses. The reasoning for making this act like a regular
+  device described by DT is so the virtual device that binds the buffer
+  can be made to act as if it is out on a bus or behind an IOMMU, for
+  example.
+
+properties:
+  compatible:
+    const: ti,dma_buf_phys
+
+required:
+  - compatible
+
+examples:
+  - |
+    dma_buf_phys {
+            compatible = "ti,dma_buf_phys";
+            iommus = <&some_iommu>;
+            dma-coherent;
+    };
-- 
2.32.0

